### PR TITLE
Closes #2207: Adding helper to detect if we are running in the main process...

### LINKED ISSUE #2207: Add helper to detect if we are running in the main process
It's ridiculously complicated to do that. Let's add something to `support-utils` for that.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt [hunk 1]
```diff
@@ -79,3 +80,25 @@ fun Context.share(text: String, subject: String = getString(R.string.mozac_suppo
         false
     }
 }
+
+/**
+ *  Returns true if we are running in the main process false otherwise.
+ */
+fun Context.isMainProcess(): Boolean {
```
Comment (Amejia481) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt]: @pocmo 
My initial idea was to cache the value as we chatted, but there some [limitations with lazy ](https://youtrack.jetbrains.com/issue/KT-9686) [[2]](https://discuss.kotlinlang.org/t/lazy-extension-properties/2351) that make it tricky to do: 
```
val Context.isMainProcess : Boolean by lazy {
    // I can't access Context@this here.
}
```

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt [hunk 2]
```diff
@@ -79,3 +80,25 @@ fun Context.share(text: String, subject: String = getString(R.string.mozac_suppo
         false
     }
 }
+
```
Comment (pocmo) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt]: You could add a `var isMainProcess: Boolean? = null` here and only run the block in `isMainProcess()` once (if it is null).

Comment (Amejia481) [reply components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt]: I was a bit afraid to do that, as `var isMainProcess: Boolean? = null` will be static and shared but two processes will have different heaps , and that will be OK. I updated the PR we the change!

**Comment by Amejia481** [comment]
👍

**Pull request merged**